Classes/common/touchdb/TD_Database.m

```diff
@@ -438,24 +577,43 @@ - (BOOL)close
     _activeReplicators = nil;
 
     [_fmdbQueue close];
+    _fmdbQueue = nil;
+    _keyProviderToOpenDB = nil;
 
     _open = NO;
     _transactionLevel = 0;
+    
     return YES;
 }
 
-- (BOOL)deleteDatabase:(NSError**)outError
+- (BOOL)deleteDatabase:(NSError **)outError
 {
-    CDTLogInfo(CDTDATASTORE_LOG_CONTEXT, @"Deleting %@", _path);
     [[NSNotificationCenter defaultCenter] postNotificationName:TD_DatabaseWillBeDeletedNotification
                                                         object:self];
-    if (_open) {
-        if (![self close]) return NO;
-    } else if (!self.exists) {
-        return YES;
+
+    if (_open && ![self close]) {
```

Comment:
`[self close]`can not fail, it only returns `NO` if the db is not open and we already validates that before calling it. I wrote the code that way to keep it similar to how it was before but it does not make sense, I can simplify it to:

```
    [self close];
```